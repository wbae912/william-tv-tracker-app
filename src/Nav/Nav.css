.nav-header {
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  background-color: rgb(78, 66, 61);
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  height: 70px;
}

.nav-logo-header {
  display: flex;
  align-items: center;
  margin-left: 2%;
  width: 50%;
}

.fa-tv {
  cursor: pointer;
  display: inline-block;
  margin-right: 2%;
  color: white;
}

.logo-div {
  margin-right: 10px;
}

.nav-title {
  display: inline-block;
  cursor: pointer;
  font-family: 'Esteban', serif;
  color: white;
  text-decoration: none;
  font-size: 20px;
}

.nav-links {
  list-style: none;
}

.nav-li {
  display: inline-block;
  padding: 0;
  padding-right: 10px;
  transition: all 0.3s ease 0s;
  font-weight: bold;
  color: white;
  font-size: 12px;
  /* CHANGE FONT SIZE WHEN LARGER SCREEN */
}

.nav-links li:hover {
  color: rgb(133, 70, 133);
}

@media only screen and (min-width: 500px) {
  .nav-li {
    font-size: 14px;
    padding-right: 20px;
  }
}

@media only screen and (min-width: 750px) {
  .nav-li {
    font-size: 16px;
    padding-right: 20px;
  }
  .nav-title {
    font-size: 24px;
  }
}

@media only screen and (min-width: 900px) {
  .nav-li {
    padding-right: 25px;
  }
}

@media only screen and (min-width: 1300px) {
  .nav-li {
    font-size: 17px;
    padding-right: 30px;
  }
}