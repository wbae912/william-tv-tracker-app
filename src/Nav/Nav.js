import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';
import './Nav.css';

export default class Nav extends Component {
  render() {
    return (
      <header className="nav-header">
        <div className="nav-logo-header">
          <div className="logo-div">
            <Link to='/'><i className="fa fa-tv fa-2x"></i></Link>
          </div>
          <Link to='/'><h2 className="nav-title">TV Tracker</h2></Link>
        </div>
        <nav className="nav">
          <ul className="nav-links">
            <NavLink to='/login' activeClassName="selected"><li className="nav-li">Login</li></NavLink>
            <NavLink to='/register' activeClassName="selected"><li className="nav-li register-li">Register</li></NavLink>
          </ul>
        </nav>
      </header>
    )
  }
}