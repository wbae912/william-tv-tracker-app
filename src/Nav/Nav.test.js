import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

describe.only('Nav component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><Nav /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Nav /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
})