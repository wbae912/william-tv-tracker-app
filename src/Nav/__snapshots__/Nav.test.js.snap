// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders UI as expected 1`] = `
<header
  className="nav-header"
>
  <div
    className="nav-logo-header"
  >
    <div
      className="logo-div"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-tv fa-2x"
        />
      </a>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <h2
        className="nav-title"
      >
        TV Tracker
      </h2>
    </a>
  </div>
  <nav
    className="nav"
  >
    <ul
      className="nav-links"
    >
      <a
        aria-current={null}
        className="nav-li"
        href="/login"
        onClick={[Function]}
      >
        <li
          className="login-li"
        >
          Login
        </li>
      </a>
      <a
        aria-current={null}
        className="nav-li"
        href="/register"
        onClick={[Function]}
      >
        <li
          className="register-li"
        >
          Register
        </li>
      </a>
    </ul>
  </nav>
</header>
`;
